common/variant/src/main/java/org/apache/spark/types/variant/Variant.java

```diff
@@ -316,6 +334,22 @@ static void toJsonImpl(byte[] value, byte[] metadata, int pos, StringBuilder sb,
       case BINARY:
         appendQuoted(sb, Base64.getEncoder().encodeToString(VariantUtil.getBinary(value, pos)));
         break;
+      case YEAR_MONTH_INTERVAL:
+        IntervalFields ymFields = VariantUtil.getYearMonthIntervalFields(value, pos);
+        int ymValue = (int) VariantUtil.getLong(value, pos);
+        appendQuoted(sb, YearMonthIntervalUtils
+                .toYearMonthIntervalANSIString(ymValue, ymFields.startField, ymFields.endField));
+        break;
+      case DAY_TIME_INTERVAL:
+        IntervalFields dtFields = VariantUtil.getDayTimeIntervalFields(value, pos);
+        long dtValue = VariantUtil.getLong(value, pos);
+        try {
```

Comment:
Why is this block in a `try` statement, but the year-month code is not?

Reply:
`toDayTimeIntervalANSIString` can throw an exception while `toYearMonthIntervalANSIString` cannot.

Reply:
Why `toDayTimeIntervalANSIString` can throw an exception? I don't think it can. If it really can, maybe need some explanation and comments on the function itself.

Reply:
It can throw an exception when the field ranges are out of bounds. That being said, `getDayTimeIntervalFields` ensures that they are always within bounds the try block is unnecessary. Thanks!